### PR TITLE
Reduce indentation in daemonset controller code

### PR DESCRIPTION
**Pull request by lucming**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by alculquicondor** [comment]
/ok-to-test

/re-title Improve indentation in daemonset controller code

**Comment by alculquicondor** [comment]
/retitle Reduce indentation in daemonset controller code

**Comment by lucming** [comment]
Is there anything else I should do?

**Comment by alculquicondor** [comment]
Find an ONWER to approve https://github.com/kubernetes/kubernetes/pull/111207#issuecomment-1186857315

**Comment by lucming** [comment]
pinging @smarterclayton for approval

File: pkg/controller/daemon/daemon_controller.go [hunk 1]
```diff
@@ -1130,28 +1130,30 @@ func (dsc *DaemonSetsController) updateDaemonSetStatus(ctx context.Context, ds *
 
 		if shouldRun {
 			desiredNumberScheduled++
-			if scheduled {
```
Comment (soltysh) [inline pkg/controller/daemon/daemon_controller.go:1133]: You could go one step further with your optimizations, this if condition has an else, which you could turn into:
```golang
		if !shouldRun {
			if scheduled {
				numberMisscheduled++
			}
			continue
		}
```
above this current `if shouldRun {...` condition, this way you'd be able to get rid of yet another indentation level. 

Feel free to open a followup and ping me on slack for review.

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**